test/e2e/test/elasticsearch/builder.go

```diff
@@ -447,13 +447,38 @@ func (b Builder) WithMutatedFrom(builder *Builder) Builder {
 func (b Builder) WithEnvironmentVariable(name, value string) Builder {
 	for i, nodeSet := range b.Elasticsearch.Spec.NodeSets {
 		for j, container := range nodeSet.PodTemplate.Spec.Containers {
-			container.Env = append(container.Env, corev1.EnvVar{Name: name, Value: value})
+			var update bool
+			for k, e := range container.Env {
+				if e.Name == name {
+					container.Env[k].Value = value
+					update = true
+				}
+			}
+			if !update {
+				container.Env = append(container.Env, corev1.EnvVar{Name: name, Value: value})
+			}
 			b.Elasticsearch.Spec.NodeSets[i].PodTemplate.Spec.Containers[j].Env = container.Env
 		}
 	}
 	return b
 }
 
+// WithPreStopAdditionalWaitSeconds allows speeding up the test by shortening the pre-stop hook runtime.
```

Comment:
```suggestion
// WithPreStopAdditionalWaitSeconds updates the PRE_STOP_ADDITIONAL_WAIT_SECONDS environment variable in the Elasticsearch container.
// It can be used to speed up the test by shortening the pre-stop hook runtime.
```